packages-next/keystone/src/scripts/build/build.ts

```diff
@@ -23,21 +42,52 @@ export function serializePathForImport(path: string) {
 export const formatSource = (src: string, parser: 'babel' | 'babel-ts' = 'babel') =>
   prettier.format(src, { parser, trailingComma: 'es5', singleQuote: true });
 
-const reexportKeystoneConfig = async (projectAdminPath: string) => {
+const reexportKeystoneConfig = async (projectAdminPath: string, isDisabled?: boolean) => {
+  if (isDisabled) {
+    // Nuke any existing files in our target directory
+    await fs.remove(projectAdminPath);
+  }
+
   // We re-export the Keystone config file into the Admin UI project folder
   // so that when we run the build step, we will end up with a compiled version
   // of the configuration file in the .next/ directory. Even if we're not building
   // an Admin UI, we still need to run the `build()` function so that this config
   // file is correctly compiled.
-  const outputDir = Path.join(projectAdminPath, 'pages', 'api');
-  const pathToConfig = Path.relative(outputDir, CONFIG_PATH);
-  const file = `
-    export { default as config } from ${serializePathForImport(pathToConfig)}
-    export default function (req, res) {
-      return res.status(500)
-    }`;
-
-  await fs.outputFile(Path.join(outputDir, '__keystone_api_build.js'), formatSource(file));
+  const pkgDir = Path.dirname(require.resolve('@keystone-next/admin-ui/package.json'));
+  const files: AdminFileToWrite[] = [
+    {
+      mode: 'write',
+      src: `export { default as config } from ${serializePathForImport(
+        Path.relative(Path.join(projectAdminPath, 'pages', 'api'), CONFIG_PATH)
+      )}
+            export default function (req, res) { return res.status(500) }`,
+      outputPath: Path.join('pages', 'api', '__keystone_api_build.js'),
+    },
+  ];
+  if (isDisabled) {
+    // These are the basic files required to have a valid Next.js project. If the
+    // Admin UI is disabled then we need to do this ourselves here.
+    files.push(
+      ...[
+        {
+          mode: 'copy' as const,
+          inputPath: Path.join(pkgDir, 'static', 'next.config.js'),
+          outputPath: 'next.config.js',
+        },
+        {
+          mode: 'copy' as const,
+          inputPath: Path.join(pkgDir, 'static', 'tsconfig.json'),
+          outputPath: 'tsconfig.json',
+        },
+        {
+          mode: 'write' as const,
+          src: `import React from 'react'; export default () => <></>`,
+          outputPath: Path.join('pages', 'index.js'),
+        },
```

Comment:
```suggestion
```

I just tested this locally without the generation of this file and it works so I don't think it's necessary.

Also a nit: `files.push(...[stuff,stuff])` -> `files.push(stuff,stuff)`

Reply:
Hmm, I had issues without this file. Let me double check 🤔 

nit: 👍

Reply:
OK, have confirmed that `index.js` isn't needed 👍